Lib/distutils/dist.py

```diff
@@ -9,10 +9,7 @@
 import re
 from email import message_from_file
 
-try:
-    import warnings
-except ImportError:
-    warnings = None
+import warnings
```

Comment:
It’s possible that the import was how it was because distutils is used to build CPython’s own extension modules.  In doubt, let’s not change this.

Reply:
Note that distutils also has its own PEP-282-style logging system, which may or may not easier to use here.  The output is controlled by verbose/quiet options passed to setup.py rather than the interpreter warning configuration system, so testing with pip would be needed to make sure that package authors would see the messages by default.

Reply:
The try/except dates back to Python 1.5 + 2.0 days.  It was added because those versions didn't have 'warnings' and distutils was targeting those versions as well.  I think removing it is safe but I will revert as it is a separate change from this issue.

Using the logging system sounds like a good idea.  I'll look into that.

Reply:
I have made the requested changes; please review again.

I tweaked the wording of the warning message, not sure it is the best.